### PR TITLE
Advanced Configuration Options

### PR DESCRIPTION
**Pull request by neverabsolute**
- can set `MEMORY_GROWTH` to any value to have Tensorflow use memory growth
- can set `MEMORY_ALLOCATED` to any float to have Tensorflow restrict the memory usage as a ratio of the total available video memory

### CODE REVIEW

**Comment by neverabsolute** [comment]
Without memory growth and default weights:
- 12GB VRAM taken up on a NVIDIA Tesla T4
- running a concurrent Pytorch model (~2-3GB VRAM needed) CUDA OOMs

With memory growth and default weights:
- 1.5GB VRAM taken up on a NVIDIA Tesla T4
- running a concurrent Pytorch model (~2-3GB VRAM needed) works fine with plenty of headroom

**Comment by neverabsolute** [comment]
By default this doesn't actually affect existing users, this just adds options for more advanced users.

File: keras_ocr/recognition.py [hunk 1]
```diff
@@ -8,7 +8,8 @@
 import numpy as np
 import cv2
 
-from . import tools
+from . import tools, config
+config.configure()
```
Comment (faustomorales) [inline keras_ocr/recognition.py:12]: I'm a little hesitant to have a side effect like this occur as part of just importing the module. Can you confirm that it is necessary to do this here and that we can't make it something people call at the top of their own code? That way, if folks are already doing their own config and using these environment variables their own way, this won't result in unexpected behavior. So the README would have a section saying something like:

> To apply these environment variables, do `keras_ocr.config.configure()`

**Review by faustomorales — COMMENTED** [review]
Thanks so much for contributing to the project! I have a question about whether we can make the use of the `configure` function optional. But otherwise, this looks great!

File: keras_ocr/recognition.py (hunk 1 above)
Comment (neverabsolute) [reply keras_ocr/recognition.py:12]: I will check it out, thanks for the feedback.

**Comment by neverabsolute** [comment]
That does work as expected, and I have implemented it. Thank you for the suggestion!

**Comment by faustomorales** [comment]
Thanks!

**Comment by faustomorales** [comment]
This new configuration code is now included in v0.9.0 of the package on PyPi. Thanks, again!

**Comment by neverabsolute** [comment]
Happy to help, this is huge for the work that I'm doing right now.

**Pull request merged**